playbook_snapshot/lib/src/snapshot_device.dart

```diff
@@ -1,30 +1,14 @@
+import 'package:flutter/cupertino.dart';
 import 'package:flutter/rendering.dart';
 
-enum SnapshotDevice {
-  iPhoneSE,
-  iPhone8,
-  iPhone11,
-  pixel4,
-  xperia,
-}
-
-extension SnapshotDeviceExt on SnapshotDevice {
-  Size get size {
-    switch (this) {
-      case SnapshotDevice.iPhoneSE:
-        return const Size(320, 568);
-      case SnapshotDevice.iPhone8:
-        return const Size(375, 667);
-      case SnapshotDevice.iPhone11:
-        return const Size(414, 896);
-      case SnapshotDevice.pixel4:
-        return const Size(411, 869);
-      case SnapshotDevice.xperia:
-        return const Size(360, 640);
-    }
-  }
+class SnapshotDevice {
+  const SnapshotDevice({
+    required this.name,
+    required this.size,
+    required this.platform,
+  });
```

Comment:
For backward compitability, we can still keep some predefined devices via `static final` field:

```dart
static final iphoneSE = SnapshotDevice(...);
```